lib/models/obms.js

```diff
@@ -106,6 +108,37 @@ function ObmsModelFactory (
             next();
         },
 
+        publishEvent: function(nodeId, next) {
+            return waterline.nodes.findByIdentifier(nodeId)
```

Comment:
Is your thought that the .catch() on line 125 would not be able to handle an exception thrown in the promise chain?